lms/services/vitalsource/factory.py

```diff
@@ -4,9 +4,14 @@
 
 def service_factory(_context, request):
     settings = request.find_service(name="application_instance").get_current().settings
-    api_key = request.registry.settings["vitalsource_api_key"]
+
+    customer_key = settings.get("vitalsource", "api_key")
 
     return VitalSourceService(
-        client=VitalSourceClient(api_key) if api_key else None,
+        # It's important to pass None here if there's no key, so the service
+        # knows it's been disabled. The client will raise an error anyway if
+        # you try and create one with no API key.
+        client=VitalSourceClient(customer_key) if customer_key else None,
+        user_lti_param=settings.get("vitalsource", "user_lti_param"),
```

Comment:
Do we need to default here to "user_id" or similar? Or is that the job of the admin pages when enabling VS support?

Reply:
We don't know that there's any kind of standard yet. Until we see a few we don't really know what the fields are.